### DISCUSSION TITLE
Alarms in asyncua

### DISCUSSION THREAD

**Discussion by josa8011** [FreeOpcUa/opcua-asyncio#918] [Q&A]
I'm using your python library for OpcUa and need alarms for my application.

I've tried following the discussions in https://github.com/FreeOpcUa/opcua-asyncio/pull/365 and https://github.com/FreeOpcUa/opcua-asyncio/pull/112 but can't exactly figure out if alarms are supported or not.

I took some code from an example in one of the pull requests and the code runs but I am not sure what is wrong with it. Whenever I try to generate an alarm the server prints a long message ending with: does not fit WhereClause, not generating event.

So my question is, can I generate alarms with asyncua and if so, are there any pointers on how to do it somewhere?

**Comment by schroeder-** [comment]
Do you need alarms on the client side or the server side? 
For the client side support is possible with some extra steps.

**Reply by josa8011** [reply]
Thanks for your reply and sorry for not clarifying. I need alarms on the server side.

**Comment by josa8011** [comment]
@Koseng @JoeyFaulkner I see you were very involved in the pull requests I mentioned. Do you have any insights into this?
